Easydict/Feature/ViewController/Window/BaseQueryWindow/EZBaseQueryViewController.m

```diff
@@ -245,13 +244,31 @@ - (void)dealloc {
 - (void)handleServiceUpdate:(NSNotification *)notification {
     NSDictionary *userInfo = notification.userInfo;
     EZWindowType type = [userInfo[EZWindowTypeKey] integerValue];
+    NSString *serviceType = [notification.userInfo objectForKey:EZServiceTypeKey];
+    if ([serviceType length] != 0 && type == self.windowType) {
+        [self updateSingleService:serviceType windowType:type];
+        return;
+    }
     if (type == self.windowType || !userInfo) {
-        [self updateServices];
+        [self updateServices:[EZLocalStorage.shared allServices:self.windowType]];
+    }
+}
+
+- (void)updateSingleService:(NSString *)serviceType windowType:(EZWindowType)windowType {
+    NSMutableArray<EZQueryService *> *newServices = [[NSMutableArray alloc] init];
+    for (EZQueryService *service in self.services) {
+        if ([service serviceType] == serviceType) {
+            EZQueryService *updatedService = [EZLocalStorage.shared service:serviceType windowType:windowType];
+            [newServices addObject:updatedService];
+        } else {
+            [newServices addObject:service];
+        }
     }
+    [self updateServices:newServices];
```

Comment:
In addition, this method looks like it can update one service alone, but in fact it ends up updating all services.

Reply:
Yes, the function name maybe confusing, this actually did update one single service in current service list and then pass the new service array to `updateServices `